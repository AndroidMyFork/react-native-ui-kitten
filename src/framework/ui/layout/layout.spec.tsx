import React from 'react';
import {
  render,
  shallow,
  RenderAPI,
} from 'react-native-testing-library';
import {
  styled,
  StyleProvider,
  StyleProviderProps,
} from '@kitten/theme';
import {
  Layout,
  LayoutProps,
} from './layout.component';
import * as config from './layout.spec.config';

const StyledComponent = styled<Layout, LayoutProps>(Layout);

const Mock = (props?: LayoutProps): React.ReactElement<StyleProviderProps> => (
  <StyleProvider mapping={config.mapping} theme={config.theme} styles={{}}>
    <StyledComponent {...props} />
  </StyleProvider>
);

const renderComponent = (props?: LayoutProps): RenderAPI => render(<Mock {...props}/>);

describe('@layout: matches snapshot', () => {

  it('default', () => {
    const component = renderComponent();
    const { output } = shallow(component.getByType(Layout));

    expect(output).toMatchSnapshot();
  });

  it('with styles', () => {
    const component = renderComponent({ style: { height: 300 }});
    const { output } = shallow(component.getByType(Layout));

    expect(output).toMatchSnapshot();
  });

});
