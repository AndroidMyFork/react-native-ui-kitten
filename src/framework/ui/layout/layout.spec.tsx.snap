// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@layout: matches snapshot default 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderColor": "#3366FF",
        "borderRadius": 3,
        "borderWidth": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
      undefined,
    ]
  }
/>
`;

exports[`@layout: matches snapshot with styles 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderColor": "#3366FF",
        "borderRadius": 3,
        "borderWidth": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
      Object {
        "height": 300,
      },
    ]
  }
/>
`;
